tgui/packages/tgui/interfaces/ChemDispenser.js

```diff
@@ -0,0 +1,439 @@
+        </Button>
+      }
+      buttons={(
+        <Box align="left" as="span">
+          {"Remove Amount: "}
+          <NumberInput
+            width={4}
+            format={value => value + "u"}
+            value={removeAmount}
+            minValue={1}
+            maxValue={100}
+            onDrag={(e, value) => setRemoveAmount(value)} />
+        </Box>
+      )}>
+      <Table.Row>
+        <Table.Cell bold collapsing textAlign="center" />
+        <Table.Cell collapsing />
+      </Table.Row>
+      <Box color="label">
+        {!beakerContents.length && (
+          "No Contents"
+        )}
+      </Box>
+      {beakerContents.map((reagent, indexContents) => (
+        <Table.Row key={indexContents}>
+          <Table.Cell collapsing
+            textAlign="left"
+            position="relative">
+            <Icon
+              pr={stateMap[reagent.state].pr}
+              style={{
+                "text-shadow": "0 0 3px #000;",
+              }}
+              color={"rgba(" + reagent.colorR + "," + reagent.colorG + ", " + reagent.colorB + ", 1)"}
+              name={iconToggle ? stateMap[reagent.state].icon : "circle"} />
+            <Tooltip.Overflow width="225px"
+              content={" ( " + reagent.volume + "u ) " + reagent.name}
+              style={{
+                "line-height": "15px",
+              }}>
+              {" ( " + reagent.volume + "u ) " + reagent.name }
+            </Tooltip.Overflow>
+          </Table.Cell>
+          <Table.Cell collapsing textAlign="left">
+            <Box mt={0.5}>
+              <Button
+                icon="filter"
+                onClick={() => act("isolate", {
+                  reagentId: reagent.id,
+                })}>
+                Isolate
+              </Button>
+              <Button
+                icon="minus"
+                onClick={() => act("all", {
+                  amount: removeAmount, reagentId: reagent.id,
+                })}>
+                All
+              </Button>
+              {removeReagentButtons.map((amount, indexButtons) => (
+                <Button
+                  key={indexButtons}
+                  icon="minus"
+                  onClick={() => act("remove", {
+                    amount: amount, reagentId: reagent.id,
+                  })}>
+                  {amount}
+                </Button>
+              ))}
+            </Box>
+          </Table.Cell>
+        </Table.Row>))}
+    </Section>
+  );
+};
+
+
+
+
+export const BeakerContentsGraph = (props, context) => {
+  const { data } = useBackend(context);
+  const [sort, setSort] = useSharedState(context, 'sort', 1);
+  const {
+    beakerContents,
+    maximumBeakerVolume,
+    beakerTotalVolume,
+  } = data;
+  const finalColor = data.finalColor || "";
+  const sortMap = [
+    {
+      id: 0,
+      icon: "sort-amount-down",
+      contents: "",
+      algorithm: (a, b) => b.volume - a.volume,
```

Comment:
Probably call these `comparator` or `compareFunction` rather than `algorithm`, to give some meaning as to what it does. Naming convention for the native ES6 sort method is `compareFunction` ([link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)).